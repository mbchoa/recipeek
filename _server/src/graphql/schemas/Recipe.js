export const typeDef = `
  type Recipe {
    calories: Float,
    dietLabels: [String],
    digest: [Nutrient],
    healthLabels: [String],
    image: String,
    ingredients: [Ingredient],
    ingredientLines: [String],
    label: String,
    source: String,
    totalWeight: Float,
    url: String,
    yield: Int,
  }
`;
