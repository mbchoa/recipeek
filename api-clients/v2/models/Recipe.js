import t from 'tcomb';
import { assign, map } from 'lodash';

import DietLabels from './DietLabels';
import HealthLabels from './HealthLabels';
import Ingredient from './Ingredient';
import Nutrient from './Nutrient';

const Recipe = t.struct({
  calories: t.Number,
  dietLabels: t.list(DietLabels),
  healthLabels: t.list(HealthLabels),
  imageUrl: t.String,
  ingredients: t.list(Ingredient),
  name: t.String,
  nutrients: t.list(Nutrient),
  source: t.String,
  sourceUrl: t.String,
  yield: t.Number,
}, 'Recipe');

Recipe.of = t.func(
  [t.Object],
  Recipe,
  'Recipe'
).of(({ recipe }) => {
  const {
    image: imageUrl,
    ingredients,
    label: name,
    totalNutrients,
    url: sourceUrl,
  } = recipe;

  return assign(  
    {},
    recipe,
    {
      imageUrl,
      name,
      sourceUrl,
    },
    {
      ingredients: map(ingredients, ingredient => new Ingredient(ingredient)),
      nutrients: map(totalNutrients, nutrient => new Nutrient(nutrient)),
    }
  );
});

export default Recipe;
