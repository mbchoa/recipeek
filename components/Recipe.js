import React from 'react';

import { titleCase } from '../helpers/string';

const preStyle = {
  backgroundColor: 'light-gray',
};

const recipeImgStyle = {
  height: 300,
};

const keywordStyle = {
  fontSize: 14,
};

const Recipe = ({ 
  index,
  data 
}) =>
  <div className="thumbnail">
    <a href={data.source_url}>
      <img style={recipeImgStyle} src={data.image_url}/>
    </a>
    <div className="caption">
      <h5>{titleCase(data.title)}</h5>
      <h6>by <a href={data.publisher_url}>{data.publisher}</a></h6>
      <div className="keywords">
        {data.keywords.map(word =>
          <span style={keywordStyle} className="label label-pill label-info">{word}</span>
        )}
      </div>
    </div>
  </div>

export default Recipe;
