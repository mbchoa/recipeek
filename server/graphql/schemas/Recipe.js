export default `
  type Recipe {
    calories: Float,
    dietLabels: [String],
    healthLabels: [String],
    image: String,
    ingredients: [Ingredient],
    label: String,
    source: String,
    totalDaily: [Nutrient],
    totalNutrients: [Nutrient],
    totalWeight: Float,
    uri: String,
    url: String,
    yield: Int,
  }
`;
