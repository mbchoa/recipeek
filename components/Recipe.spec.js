import React from 'react';
import { shallow } from 'enzyme';

import Recipe from './Recipe';

describe('Recipe', () => {
    function renderComponent(props) {
        return shallow(
            <Recipe { ...props } />
        );
    }

    it('should render Recipe component', () => {
        expect(renderComponent({
            healthLabels: [ 'stinky', 'sour' ],
            imageUrl: 'www.stinkyfruit.com',
            name: 'stinky fruit',
            source: 'Stinky Fruit Blog',
            sourceUrl: 'www.strinkyfruit.com/source'
        })).toMatchSnapshot();
    });
});
