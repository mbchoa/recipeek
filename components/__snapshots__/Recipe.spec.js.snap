// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should render Recipe component 1`] = `
<div
  className="thumbnail"
>
  <a
    href="www.strinkyfruit.com/source"
  >
    <img
      alt="stinky fruit"
      src="www.stinkyfruit.com"
      style={
        Object {
          "height": 300,
        }
      }
    />
  </a>
  <div
    className="caption"
  >
    <h5>
      Stinky Fruit
    </h5>
    <h6>
      by 
      Stinky Fruit Blog
    </h6>
    <div
      className="keywords"
    >
      <span
        className="label label-pill label-info"
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        stinky
      </span>
      <span
        className="label label-pill label-info"
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        sour
      </span>
    </div>
  </div>
</div>
`;
